scripts/report-bundle-sizes-as-a-pr-comment.js

```diff
@@ -0,0 +1,189 @@
+const { runMain, timeout } = require('./lib/execution-utils')
+const { getOrg2ApiKey, getGithubAccessToken, getOrg2AppKey } = require('./lib/secrets')
+
+const PR_COMMENT_HEADER = 'Bundles Sizes Evolution'
+const BASE_BRANCH = process.env.MAIN_BRANCH
+const LOCAL_BRANCH = process.env.CI_COMMIT_REF_NAME
+const PR_COMMENTER_AUTH_TOKEN = command`authanywhere`.run().split(' ')[2].trim()
+const GITHUB_TOKEN = getGithubAccessToken()
+const ONE_DAY_IN_SECOND = 24 * 60 * 60
+const TIMEOUT_DURATION_MS = 5000
+const FETCH_RETRIES = 4
+
+runMain(async () => {
+  const lastCommonCommit = getLastCommonCommit(BASE_BRANCH, LOCAL_BRANCH)
+  const latestLocalCommit = process.env.CI_COMMIT_SHORT_SHA
+  const prs = await getPRs(LOCAL_BRANCH)
+  if (prs.length === 0) {
+    console.log('No pull requests found for the branch')
+    process.exit(0)
+  }
+  if (prs.length > 1) {
+    console.log('Multiple pull requests found for the branch')
+    process.exit(0)
+  }
+  const mainBranchBundleSizes = await fetchAllPackagesBundleSize(loadQuery(lastCommonCommit))
+  const currentBranchBundleSizes = await fetchAllPackagesBundleSize(loadQuery(latestLocalCommit))
+  const difference = compare(mainBranchBundleSizes, currentBranchBundleSizes)
+  const commentId = await retrieveExistingCommentId(prs[0].number)
+  await updateOrAddComment(difference, mainBranchBundleSizes, currentBranchBundleSizes, prs[0].number, commentId)
+})
+
+function getLastCommonCommit(baseBranch) {
+  try {
+    command`git fetch origin`.run()
+    const commandOutput = command`git merge-base origin/${baseBranch} HEAD`.run()
+    // SHA commit is truncated to 8 caracters as bundle sizes commit are exported in short format to logs for convenience and readability.
+    return commandOutput.trim().substring(0, 8)
+  } catch (error) {
+    throw new Error('Failed to get last common commit', { cause: error })
+  }
+}
+
+async function getPRs(localBranch) {
+  const response = await fetch(`https://api.github.com/repos/DataDog/browser-sdk/pulls?head=DataDog:${localBranch}`, {
+    method: 'GET',
+    headers: {
+      Authorization: `token ${GITHUB_TOKEN}`,
+    },
+  })
+  if (!response.ok) {
+    throw new Error(`HTTP Error Response: ${response.status} ${response.statusText}`)
+  }
+  return response.body ? response.json() : null
+}
+
+function loadQuery(commitSha) {
+  return [
+    {
+      name: 'bundles_sizes_rum',
+      queries: `avg:bundle_sizes.rum{commit:${commitSha}}`,
+    },
+    {
+      name: 'bundles_sizes_logs',
+      queries: `avg:bundle_sizes.logs{commit:${commitSha}}`,
+    },
+    {
+      name: 'bundles_sizes_rum_slim',
+      queries: `avg:bundle_sizes.rum_slim{commit:${commitSha}}`,
+    },
+    {
+      name: 'bundles_sizes_worker',
+      queries: `avg:bundle_sizes.worker{commit:${commitSha}}`,
+    },
+  ]
+}
+
+function fetchAllPackagesBundleSize(bundleSizeQueries) {
+  return Promise.all(bundleSizeQueries.map((budget) => fetchBundleSizesMetrics(budget)))
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏not sure to what you want to express with `budget` naming, could we clarify that?

Reply:
https://github.com/DataDog/web-ui/blob/a58062ac126efaac22015b80dc798e84cf15d69c/packages-to-migrate/budget-plan/src/drivers/datadog/index.ts#L447

I inspired myself from the budget plan, they call it budget. I thought that was the appropriate name

Reply:
oh ok, but FMU there they actually have a "budget" with success/warning/alert thresholds.
Here, we are just display the difference without try to evaluate this result, at least for now.
So budget does not really seem appropriate.

Reply:
counterproposal: what about using `metric` naming?

```js
function buildMetrics(commitSha) {
  return [
    {
      name: 'bundles_sizes_rum',
      query: `avg:bundle_sizes.rum{commit:${commitSha}}`,
    },
    ...
  ]
}

fetchAllPackagesBundleSize(metrics) {
  return Promise.all(bundleSizeQueries.map((metric) => fetchBundleSizesMetric(metric)))
}

async function fetchBundleSizesMetric(metric) {
  ...
  const response = await fetch(
      `https://api.datadoghq.com/api/v1/query?query=${metric.query}&from=${date}&to=${now}`,
  ...
  return {
    name: metric.name,
    size: ...,
  }
}
```

with:

- updating the data structure `{name, queries}` to `{name, query}`, unless I am missing the reason why it should be plural
- renaming `fetchBundleSizesMetrics` to `fetchBundleSizesMetric`, since we are fetching a single metric value by call

would it make sense?

Reply:
I am not sure to understand, the purpose of the first function is to set queries with their respective sha. Why not buildQueryMetrics ?